src/hotspot/share/gc/g1/g1CollectionSet.cpp

```diff
@@ -331,19 +330,16 @@ static int compare_region_idx(const uint a, const uint b) {
   return static_cast<int>(a-b);
 }
 
-// The current mechanism skips evacuation of pinned old regions like g1 does for
-// young regions:
-// * evacuating pinned marking collection set candidate regions (available during mixed
-//   gc) like young regions would not result in any memory gain but only take additional
-//   time away from processing regions that would actually result in memory being freed.
-//   To advance mixed gc progress (we committed to evacuate all marking collection set
-//   candidate regions within the maximum number of mixed gcs in the phase), move them
-//   to the optional collection set candidates to reclaim them asap as time permits.
-// * evacuating out retained collection set candidates would also just take up time with
-//   no actual space freed in old gen. Better to concentrate on others.
-//   Retained collection set candidates are aged out, ie. made to regular old regions
-//   without remembered sets after a few attempts to save computation costs of keeping
-//   them candidates for very long living pinned regions.
+// The current mechanism for evacuting pinned old regions is a below:
+// * pinned regions in marking collection set candidate (available during mixed gc) are evacuated like
+//   pinned young regions to avoid complexity of dealing with pinned regions that are part of a
+//   collection group sharing a single cardset. These regions will partially evacuated and added to the
+//   retained collection set by the handle evacuation failure mechanism.
+// * evacuating pinned regions out of retained collection set candidates would also just take up time
+//   with no actual space freed in old gen. Better to concentrate on others. So we skipped over pinned
+//   regions in retained collection set candidates Retained collection set candidates are aged out, ie.
+//   made to regular old regions without remembered sets after a few attempts to save computation costs
+//   of keeping them candidates for very long living pinned regions.
```

Comment:
```suggestion
// The current mechanism for evacuating pinned old regions is as below:
// * pinned regions in the marking collection set candidate list (available during mixed gc) are evacuated like
//   pinned young regions to avoid the complexity of dealing with pinned regions that are part of a
//   collection group sharing a single cardset. These regions will be partially evacuated and added to the
//   retained collection set by the evacuation failure handling mechanism.
// * evacuating pinned regions out of retained collection set candidates would also just take up time
//   with no actual space freed in old gen. Better to concentrate on others. So we skip over pinned
//   regions in retained collection set candidates. Retained collection set candidates are aged out, ie.
//   made to regular old regions without remembered sets after a few attempts to save computation costs
//   of keeping them candidates for very long living pinned regions.
```